.github/actions/publish/action.yml

```
@@ -0,0 +1,87 @@
+---
+name: publish binaries
+description: package GitHub release for node-pre-gyp
+# https://github.com/product-os/flowzone/tree/master/.github/actions
+inputs:
+  json:
+    description: 'JSON stringified object containing all the inputs from the calling workflow'
+    required: true
+  secrets:
+    description: 'JSON stringified object containing all the secrets from the calling workflow'
+    required: true
+  variables:
+    description: 'JSON stringified object containing all the variables from the calling workflow'
+    required: true
+  # --- custom environment
+  NODE_VERSION:
+    description: 'Node.js version'
+    default: '18.x'
+
+runs:
+  # https://docs.github.com/en/actions/creating-actions/creating-a-composite-action
+  using: 'composite'
+  steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.NODE_VERSION }}
+        registry-url: 'https://registry.npmjs.org'
+
+    - name: Build JSON array from pipe separated list
+      id: config
+      uses: kanga333/json-array-builder@c7cd9d3a8b17cd368e9c2210bc3c16b0e2714ce5 # v0.2.1
+      with:
+        cmd: bash -c "echo $INPUT | tr -d '[:space:]'"
+        separator: "|"
+      env:
+        INPUT: ${{ env.matrix_value }}
+
+    - name: Login to GitHub Container Registry
+      continue-on-error: true
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ fromJSON(inputs.secrets).GITHUB_TOKEN }}
+
+    - name: Copy the binary from the built docker image
+      shell: bash
+      run: |
+        set -ex
+
+        image="ghcr.io/balena-io-modules/node-systemd:${{ github.event.pull_request.head.sha }}"
```

```suggestion
        image="ghcr.io/balena-io-modules/node-systemd:build-${{ github.event.pull_request.head.sha }}"
```
I prefer if we use the `build-` prefix when referencing the commit sha. I regret publishing the former as part of the Flowzone publish job, but we can't easily remove them now. Both formats are available.